### PR TITLE
Reference issue numbers in the `TODO`s

### PR DESCRIPTION
**Pull request by mertwole**

### CODE REVIEW

File: JamGame/Assets/Scripts/Employee/EmployeeModifiers.cs [hunk 1]
```diff
@@ -130,7 +130,6 @@ public void RegisterBuff(Buff buff)
             }
         }
 
-        // TODO: match type of effect with corresponding Executor type.
```
Comment (mertwole) [inline JamGame/Assets/Scripts/Employee/EmployeeModifiers.cs:133]: Calling side working with `IEffect`, so type information must be recovered this way anywhere

File: JamGame/Assets/Scripts/Employee/Needs/Need.cs [hunk 2]
```diff
@@ -69,7 +69,6 @@ public NeedProperties Combine(NeedProperties other)
 
         public float Satisfied => satisfied;
 
-        // TODO: refactor
```
Comment (mertwole) [inline JamGame/Assets/Scripts/Employee/Needs/Need.cs:72]: Outdated

File: JamGame/Assets/Scripts/Level/Boss/Task.cs [hunk 3]
```diff
@@ -11,7 +11,6 @@
 using Sirenix.OdinInspector;
 using UnityEngine;
 
-// TODO: Move all the DTOs to their source classes.
```
Comment (mertwole) [inline JamGame/Assets/Scripts/Level/Boss/Task.cs:14]: Actually it's not important to have them near classes implementing `IDataProvide<...>`

File: JamGame/Assets/Scripts/Level/Config/InventoryRoomConfig.cs [hunk 4]
```diff
@@ -10,7 +10,6 @@ public class InventoryRoomConfig
         [SerializeField]
         private CoreModel room;
 
-        // TODO: implement mandatory room logic
```
Comment (mertwole) [inline JamGame/Assets/Scripts/Level/Config/InventoryRoomConfig.cs:13]: #175

File: JamGame/Assets/Scripts/Location/NeedProvider.cs [hunk 5]
```diff
@@ -51,7 +51,6 @@ public void Drop()
                 _ = needProvider.waitingLine.Remove(this);
             }
 
-            // TODO: private
```
Comment (mertwole) [inline JamGame/Assets/Scripts/Location/NeedProvider.cs:54]: Not important

File: JamGame/Assets/Scripts/Location/NeedProvider.cs [hunk 6]
```diff
@@ -183,7 +182,6 @@ Action releasedCallback
                 return;
             }
 
-            // TODO: Do we need to check criterias again?
```
Comment (mertwole) [inline JamGame/Assets/Scripts/Location/NeedProvider.cs:186]: No

File: JamGame/Assets/Scripts/Overlay/Manager.cs [hunk 7]
```diff
@@ -15,7 +15,6 @@ public void ApplyOverlay<O>(O overlay)
     [AddComponentMenu("Scripts/Overlay/Overlay.Manager")]
     public class Manager : MonoBehaviour, IOverlayManager
     {
-        // TODO: Update in runtime.
```
Comment (mertwole) [inline JamGame/Assets/Scripts/Overlay/Manager.cs:18]: It's actually updated, but in a not effective way

File: JamGame/Assets/Scripts/Level/Boss/Task.cs (hunk 3 above)
Comment (Kiuh) [reply JamGame/Assets/Scripts/Level/Boss/Task.cs:14]: So, better hold them in one place?

**Review by Kiuh — APPROVED** [review]

File: JamGame/Assets/Scripts/Level/Boss/Task.cs (hunk 3 above)
Comment (mertwole) [reply JamGame/Assets/Scripts/Level/Boss/Task.cs:14]: It's better to have a single approach across all the data providet DTOs, as they can be emitted from many places or consumed by many observers, so if we implement different approaches to these types of connections it'll be hard to find implementations

**Pull request merged**